### PR TITLE
feat(rpc/v0.2): starknet_addInvokeTransaction

### LINKED ISSUE #608: `starknet_addInvokeTransaction`

### PR DESCRIPTION
**Pull request by Mirko-von-Leipzig**
I have some concerns as there are no proper tests we can do for this? v0.1 implementation also has no tests for it afaict.

Closes #608

### CODE REVIEW

File: crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            .await
+            .context("Sending V0 invoke transaction to gateway")?,
+        BroadcastedInvokeTransaction::V1(v1) => context
+            .sequencer
+            .add_invoke_transaction(
+                v1.version,
+                v1.max_fee,
+                v1.signature,
+                Some(v1.nonce),
+                v1.sender_address,
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs:45]: @kkovaacs if I recall correctly, this is the new name for `contract address`?

File: crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                None,
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs:46]: @kkovaacs and `entry point selector` is now deprecated?

**Comment by kkovaacs** [comment]
BTW, the `v01` implementation does have some tests: https://github.com/eqlabs/pathfinder/blob/87173b9b44dffbc54e36979bff090a33be42895c/crates/pathfinder/src/rpc/v01.rs#L2635

File: crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs:45]: I think so, yes.

File: crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs (hunk 2 above)
Comment (kkovaacs) [reply crates/pathfinder/src/rpc/v02/method/add_invoke_transaction.rs:46]: Yes, there's no `entry_point_selector` for v1 invoke transactions (the reasoning behind it is that it's always done through an account contract, so the only meaningful entry point is `__execute__`).

**Comment by kkovaacs** [comment]
So I think we should _at least_ add the equivalents of those test cases (which are not that great, though).

**Review by CHr15F0x — APPROVED** [review]
lgtm :+1: 
iiuc the tests have already been added (?)

**Comment by Mirko-von-Leipzig** [comment]
It was confirmed that the RPC specification has bug and that `nonce` should not be a part of V0 Broadcast Invoke transactions. The chosen solution is to make the field optional, but then null it out when performing the write transaction to the gateway.

**Review by CHr15F0x — APPROVED** [review]

**Review by kkovaacs — APPROVED** [review]
LGTM

**Pull request merged**